db/osqlsqlthr.c

```diff
@@ -800,6 +809,12 @@ int osql_sock_restart(struct sqlclntstate *clnt, int maxretries,
         cheap_stack_trace();
     }
 
+    if (clnt->ctrl_sqlengine == SQLENG_NORMAL_PROCESS) {
+        osql->single_stmt_retry = 1;
+        osql->xerr.errval = (ERR_BLOCK_FAILED + ERR_VERIFY);
+        return 0;
```

Comment:
We can probably get rid of `single_stmt_retry` if we return nonzero rcode (e.g. `SQLITE_ABORT`) here. With this approach, we may also need to change `sock_restart_retryable_rcode` to not retry in this case. I think sqlinterfaces.c should be able to handle these correctly as long as we have `osql->xerr.errval` correct (see `rc_sqlite_to_client()`).

Reply:
I added the flag so that we don't print an error on a fake single-statement verify-retry (see [here](https://github.com/bloomberg/comdb2/pull/1422/files#diff-0197042d98ced8aebf540e5e208649afR1150)).

Reply:
Is it easier to return nonzero rcode here?